src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/delete_interaction_choice_set_request.cc

```diff
@@ -87,25 +94,70 @@ void DeleteInteractionChoiceSetRequest::Run() {
     return;
   }
   SendVrDeleteCommand(app);
-
-  smart_objects::SmartObject msg_params =
-      smart_objects::SmartObject(smart_objects::SmartType_Map);
-
-  msg_params[strings::interaction_choice_set_id] = choice_set_id;
-  msg_params[strings::app_id] = app->app_id();
-
-  app->RemoveChoiceSet(choice_set_id);
-
-  // Checking of HMI responses will be implemented with APPLINK-14600
-  const bool result = true;
-  SendResponse(result, mobile_apis::Result::SUCCESS);
 }
 
 bool DeleteInteractionChoiceSetRequest::Init() {
   hash_update_mode_ = HashUpdateMode::kDoHashUpdate;
   return true;
 }
 
+void DeleteInteractionChoiceSetRequest::on_event(
+    const event_engine::Event& event) {
+  using namespace helpers;
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  const smart_objects::SmartObject& message = event.smart_object();
+  response_result_code_ = static_cast<hmi_apis::Common_Result::eType>(
+      message[strings::params][hmi_response::code].asInt());
+  const std::uint32_t correlation_id = static_cast<uint32_t>(
+      message[strings::params][strings::correlation_id].asUInt());
+
+  auto found_request =
+      std::find(sent_requests_.begin(), sent_requests_.end(), correlation_id);
+  if (found_request == sent_requests_.end()) {
+    LOG4CXX_DEBUG(logger_,
+                  "Request with " << *found_request
+                                  << " correlation_id is not found.");
+    return;
+  }
+  sent_requests_.erase(found_request);
+
+  if (sent_requests_.empty()) {
+    const mobile_apis::Result::eType result_code =
+        MessageHelper::HMIToMobileResult(response_result_code_);
+    response_result_ =
+        helpers::Compare<mobile_apis::Result::eType, helpers::EQ, helpers::ONE>(
+            result_code,
+            mobile_apis::Result::SUCCESS,
+            mobile_apis::Result::WARNINGS);
+    if (response_result_) {
+      ApplicationSharedPtr app =
+          application_manager_.application(connection_key());
+      const uint32_t choice_set_id =
+          (*message_)[strings::msg_params][strings::interaction_choice_set_id]
+              .asUInt();
+      app->RemoveChoiceSet(choice_set_id);
+    }
+    SendResponse(response_result_, result_code);
```

Comment:
@IGapchuk `SendReponse` call should be always in the end of function as after this call command will be destroyed at its members could be invalidated

Reply:
@AKalinich-Luxoft Replaced in e32b004ae4d7e84e153d6b8082cb3c1a800a41ec.